pkg/core/resources/model/utils.go

```diff
@@ -96,3 +96,11 @@ func IsEmpty(spec ResourceSpec) bool {
 func PtrTo[T any](t T) *T {
 	return &t
 }
+
+func SafeDeref[T any](ptr *T) T {
+	if ptr == nil {
+		var zero T
+		return zero
+	}
+	return *ptr
+}
```

Comment:
I can't see how this is connected with our core model. I'd put it into `pkg/utils/pointers/pointers.go` and then use `pointers.To(X)` and `pointers.SafeDeref(*X)` or even `pointers.Deref` to be shorter.

Reply:
Moved to another PR https://github.com/kumahq/kuma/pull/5586